lib/frame.rb

```diff
@@ -0,0 +1,60 @@
+# knows if it's a strike, spare or over
+class Frame
+  TOTAL_PINS = 10
+
+  attr_reader :remaining_pins, :bonus, :rolls
```

Comment:
Should these really be publicly readable? Are you doing this mostly / only for testing?
It's useful when writing feature test to think about what the rest of the program will need to know about objects of this class. 
Here I would say they mostly need to:

**INPUTS**
- Create it
- Pass it a roll

**OUTPUTS**
- Know if it's over
- Know the score
- Know things like is it a strike or spare (maybe)
- Getting errors

That's all. So that's all the behaviour you need to expose and test. No testing the internal state. This way, it allows you more flexibility to change how that internal state works later on, while keeping the public interface and functionality.

Reply:
Reading more code, actually maybe rolls is needed as you show the rolls on your scorecard later.

Reply:
yeah I see what you mean, I guess they don't need to be readable. I was kind of thinking ahead to the UI, that maybe this is all info I'd want to display to the user in game. However that can also be achieved through more purpose built public methods. thanks!